core/src/test/scala/org/apache/spark/CheckpointSuite.scala

```diff
@@ -266,13 +309,44 @@ class CheckpointSuite extends SparkFunSuite with RDDCheckpointTester with LocalS
   override def sparkContext: SparkContext = sc
 
   runTest("basic checkpointing") { reliableCheckpoint: Boolean =>
-    val parCollection = sc.makeRDD(1 to 4)
-    val flatMappedRDD = parCollection.flatMap(x => 1 to x)
-    checkpoint(flatMappedRDD, reliableCheckpoint)
-    assert(flatMappedRDD.dependencies.head.rdd === parCollection)
-    val result = flatMappedRDD.collect()
-    assert(flatMappedRDD.dependencies.head.rdd != parCollection)
-    assert(flatMappedRDD.collect() === result)
+    startSparkContext()
+    testBasicCheckpoint(sc, reliableCheckpoint)
+  }
+
+  runTest("compression with snappy", skipLocalCheckpoint = true) { _: Boolean =>
```

Comment:
After you change the config to `spark.checkpoint.compress`, you don't need to test all compression codecs. Just write one test for the default codec. Others should be covered in `CompressionCodecSuite`.